src/lib/phdi-building-blocks/phdi_building_blocks/standardize.py

```diff
@@ -1,93 +1,172 @@
+    """
+    Given a FHIR bundle and a type of standardization to perform,
+    transform all names in all patients in the bundle. The default
+    standardization behavior is our defined non-numeric, space-trimming,
+    full capitalization standardization, but other modes may be specified.
+    Also, augment the patient resources with metrics information about
+    whether standardization improved the quality of any names, if
+    desired.
+    :param dict bundle: The FHIR bundle to standardize patients in
+    :param str standardization_mode: The type of standardization to perform.
+    Options include: "non_numeric_caps"
+    :param bool add_name_metrics: Whether to store tracking data about
+    if any names were changed in the patient resources
+    :param bool overwrite: Whether to overwrite the original data with
+    the transformed values. Default is yes.
+    :return: The bundle with all patients having transformed names
+    :rtype: dict
+    """
+    # Copy the data if we don't want to overwrite the origina
+    if not overwrite:
+        bundle = copy.deepcopy(bundle)
 
+    # Handle all patients individually
     for resource in find_patient_resources(bundle):
         patient = resource.get("resource")
-        if "extension" not in patient:
-            patient["extension"] = []
+        if standardization_mode == "non_numeric_caps":
+            standardize_names_for_patient(
+                patient, non_numeric_caps_standardization, add_name_metrics
+            )
+        else:
+            raise ValueError("Invalid standardization mode supplied.")
+    return bundle
 
-        # Transform names
-        for name in patient.get("name", []):
-            if "family" in name:
-                std_family = standardize(name["family"])
-                raw_family = name["family"]
-                patient["extension"].append(
-                    {
-                        "url": "http://usds.gov/fhir/phdi/StructureDefinition/family-name-was-standardized",  # noqa
-                        "valueBoolean": raw_family != std_family,
-                    }
-                )
-                name["family"] = std_family
-
-            if "given" in name:
-                std_givens = [standardize(g) for g in name["given"]]
-                raw_givens = [g for g in name["given"]]
-                any_diffs = any(
-                    [raw_givens[i] != std_givens[i] for i in range(len(raw_givens))]
-                )
-                patient["extension"].append(
-                    {
-                        "url": "http://usds.gov/fhir/phdi/StructureDefinition/given-name-was-standardized",  # noqa
-                        "valueBoolean": any_diffs,
-                    }
-                )
-                name["given"] = std_givens
 
-    return bundle
+def standardize_names_for_patient(
+    patient: dict,
+    transform_func: Callable = non_numeric_caps_standardization,
```

Comment:
We should not be passing functions into functions. All of these functions are globally accessible, and passing functions in just makes the code harder to read in my opinion. Instead, we could have a `FUNC_MAP` global variable that looked something like

```python
FUNC_MAP = {
    "non_numeric_uppercase": non_numeric_uppercase_standardization,
}
```
and then in the code we have:

```python
def standardize_names_for_patient(patient: dict, transformation: str, ...):
    transform_func = FUNC_MAP[transformation] # note that we're okay with a KeyError exception here
    <code>
```

Reply:
But again, I don't think we should be using this approach, and instead should add the various aspects of the data they can change as parameters to the function.

Reply:
I've implemented this change such that we no longer provide a named transformation, but instead allow the user to customize which transformations they'd like to apply where applicable. I believe this makes the code much more extensible and reusable, while also reducing the amount of information a user must have about our code base in order to use it.